cdap-ui/app/cdap/components/Tags/index.js

```diff
@@ -15,33 +15,123 @@
  */
 
 import React, {PropTypes, Component} from 'react';
-import {objectQuery} from 'services/helpers';
+import {MyMetadataApi} from 'api/metadata';
+import Mousetrap from 'mousetrap';
+import NamespaceStore from 'services/NamespaceStore';
+import {convertEntityTypeToApi} from 'services/entity-type-api-converter';
+import SpotlightModal from 'components/SpotlightSearch/SpotlightModal';
 require('./Tags.scss');
 import T from 'i18n-react';
 
 export default class Tags extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      tags: {
-        system: objectQuery(this.props, 'entity', 'metadata', 'metadata', 'SYSTEM', 'tags') || [],
-        user: objectQuery(this.props, 'entity', 'metadata', 'metadata', 'USER', 'tags') || []
-      }
+      systemTags: [],
+      userTags: [],
+      searchModalOpen: false,
+      showInputField: false,
+      currentInputTag: ''
     };
+
+    this.params = {
+      namespace: NamespaceStore.getState().selectedNamespace,
+      entityType: convertEntityTypeToApi(this.props.entity.type),
+      entityId: this.props.entity.id,
+    };
+
+    this.toggleSearchModal = this.toggleSearchModal.bind(this);
+    this.toggleInputField = this.toggleInputField.bind(this);
+    this.onInputTagChange = this.onInputTagChange.bind(this);
+    this.addTag = this.addTag.bind(this);
+  }
+  componentWillMount() {
+    Mousetrap.bind('enter', this.addTag);
+
+    let systemParams = Object.assign({}, this.params, { scope: 'SYSTEM' });
+    let userParams = Object.assign({}, this.params, { scope: 'USER' });
+
+    MyMetadataApi.getTags(systemParams)
+      .combineLatest(MyMetadataApi.getTags(userParams))
+      .subscribe((res) => {
+        this.setState({
+          systemTags: res[0],
+          userTags: res[1]
+        });
+      }, (err) => {
+        console.log('Error', err);
```

Comment:
We should probably surface this error. `Unable to fetch tags` or something similar to that .